marathon-gradle-plugin/src/main/kotlin/com/malinskiy/marathon/gradle/task/GenerateMarathonfileTask.kt

```diff
@@ -0,0 +1,116 @@
+package com.malinskiy.marathon.gradle.task
+
+import com.android.build.api.variant.BuiltArtifacts
+import com.malinskiy.marathon.config.Configuration
+import com.malinskiy.marathon.config.serialization.ConfigurationFactory
+import com.malinskiy.marathon.config.vendor.VendorConfiguration
+import com.malinskiy.marathon.config.vendor.android.AndroidTestBundleConfiguration
+import com.malinskiy.marathon.gradle.GradleAndroidTestBundle
+import com.malinskiy.marathon.gradle.Const
+import com.malinskiy.marathon.gradle.service.JsonService
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.listProperty
+import org.gradle.kotlin.dsl.property
+import java.io.File
+import javax.inject.Inject
+
+open class GenerateMarathonfileTask @Inject constructor(objects: ObjectFactory) : DefaultTask() {
```

Comment:
it could be `abstract class`

Reply:
or open. As long as gradle can modify the class. If I want to initialize this task from tests - I can. If I use abstract - it's not possible.